src/main/kotlin/org/rust/ide/template/postfix/stringBasedPostfixTemplates.kt

```diff
@@ -100,6 +107,34 @@ class SomePostfixTemplate(provider: RsPostfixTemplateProvider) : SimpleExprPostf
 class OkPostfixTemplate(provider: RsPostfixTemplateProvider) : SimpleExprPostfixTemplate("ok", "Ok(expr)", provider)
 class ErrPostfixTemplate(provider: RsPostfixTemplateProvider) : SimpleExprPostfixTemplate("err", "Err(expr)", provider)
 
+class WrapTypePathPostfixTemplate(provider: RsPostfixTemplateProvider) : StringBasedPostfixTemplate("wrap", "\$wrapper$<path>", RsTypeParentsSelector(), provider) {
+    override fun getTemplateString(element: PsiElement): String = example.replace("path", element.text)
+
+    override fun expandForChooseExpression(element: PsiElement, editor: Editor) {
+        val typeRef = element as? RsTypeReference ?: return
+
+        val factory = RsPsiFactory(typeRef.project)
+        val path = factory.tryCreatePath("Wrapper<${typeRef.text}>", RustParserUtil.PathParsingMode.TYPE) ?: return
+        val newTypeRef = factory.tryCreateType(path.text) ?: return
+        val inserted = typeRef.replace(newTypeRef) as? RsBaseType ?: return
+        val ptr = inserted.createSmartPointer()
+
+        val template = editor.newTemplateBuilder(inserted) ?: return
+        val name = inserted.path?.referenceNameElement ?: return
+        template.replaceElement(name, MacroCallNode(CompleteMacro()))
+        template.withResultListener {
+            if (it != TemplateResultListener.TemplateResult.Canceled) {
```

Comment:
Why did you decide to use `it != TemplateResultListener.TemplateResult.Canceled` instead of usual `== TemplateResult.Finished` (`withFinishResultListener`)? In this case, the template moves the caret if I cancel the template by pressing `Esc`

Reply:
For some reason, if I press enter in this case in the builder: `Mutex/*caret*/<u32>`, the template ends with `BrokenOff`.

Reply:
Yes, I've reproduced. The fun fact is that it is called with `BrokenOff` for `Mutex` but not for my custom types! This looks like a bug (somewhere) and I tried to investigate it but with no success :(

Reply:
I suppose that it's not such a problem. If the user canceled the template, she probably wants to revert the wrapping anyway, so she will invoke Ctrl + Z and then it makes sense for the caret to return to the original location.

Reply:
I just don't like when something doesn't work as expected and I have no idea why